### PR TITLE
core: finalize master => main branch rename

### PR DESCRIPTION
**Pull request by paulirish**
fixes [#13925](https://github.com/GoogleChrome/lighthouse/issues/13925)

as a driveby: I fixed a bunch of github permalinks to be more recent/correct.

The changes in `core/scripts/` and `.github/workflows/` are the most interesting.

I plan to flip the setting first before landing this. Will also doublecheck the branch protection stuff.

-----------

update: 2:20pm PT Sep 27. Git branch change made.

Folks should run `git branch -m master main; git fetch origin; git branch -u origin/main main; git remote set-head origin -a` in their local checkout.

### CODE REVIEW

File: docs/recipes/lighthouse-plugin-example/readme.md [hunk 1]
```diff
@@ -11,9 +11,9 @@ Run the following to start of with the recipe as a template:
 
 ```sh
 mkdir lighthouse-plugin-example && cd lighthouse-plugin-example
-curl -L https://github.com/GoogleChrome/lighthouse/archive/master.zip | tar -xzv
-mv lighthouse-master/docs/recipes/lighthouse-plugin-example/* ./
-rm -rf lighthouse-master
+curl -L https://github.com/GoogleChrome/lighthouse/archive/main.zip | tar -xzv
+mv lighthouse-main/docs/recipes/lighthouse-plugin-example/* ./
+rm -rf lighthouse-main
```
Comment (adamraine) [inline docs/recipes/lighthouse-plugin-example/readme.md:16]: Did you confirm this is how the zip is created in github

**Review by adamraine — APPROVED** [review]
Overall LGTM

**Review by brendankenny — APPROVED** [review]
LGTM!

File: docs/recipes/lighthouse-plugin-example/readme.md (hunk 1 above)
Comment (paulirish) [reply docs/recipes/lighthouse-plugin-example/readme.md:16]: well.. i have now. :)

```sh
curl -L https://github.com/GoogleChrome/lighthouse/archive/master-to-main.zip | tar -xzv
# ...
file lighthouse-master-to-main/ # lighthouse-master-to-main/: directory
```

**Pull request merged**